devices-push/src/main/java/com/okta/devices/push/PushRemediation.kt

```diff
@@ -124,16 +132,26 @@ sealed class PushRemediation(override val challenge: PushChallenge, internal val
      *
      * @property signature Optional signature is provided to the application. This is used for constructing a CryptoObject for biometric prompt.
      */
-    class UserVerification internal constructor(challenge: PushChallenge, context: ChallengeContext, val signature: Signature? = null) : PushRemediation(challenge, context) {
+    class UserVerification internal constructor(
+        challenge: PushChallenge,
+        context: ChallengeContext,
+        val signature: Signature? = null,
+        val completionState: CompletionState? = null
+    ) : PushRemediation(challenge, context) {
         /**
          * Resolve the user verification step.
          *
          * @param authResult The [AuthenticationResult] from a successful biometric prompt.
          * @return [Result] If successful the next [Remediation] step.
          */
         fun resolve(authResult: AuthenticationResult): Result<PushRemediation> {
-            val authedCtx = ctx.copy(authResult = authResult, consentResponse = APPROVED_USER_VERIFICATION)
-            return Result.success(if (ctx.cibaEnabled)CibaConsent(challenge, authedCtx) else UserConsent(challenge, authedCtx))
+            val authedCtx =
+                if (ctx.cibaEnabled && ctx.consentResponse == APPROVED_CONSENT_PROMPT) ctx.copy(authResult = authResult)
+                else ctx.copy(authResult = authResult, consentResponse = APPROVED_USER_VERIFICATION)
+            return Result.success(
+                if (ctx.cibaEnabled && ctx.consentResponse == APPROVED_CONSENT_PROMPT && completionState != null)Completed(challenge, authedCtx, completionState)
+                else UserConsent(challenge, authedCtx)
+            )
```

Comment:
probably need to change this to
```
        suspend fun resolve(authResult: AuthenticationResult): Result<PushRemediation> {
            val authedCtx = ctx.copy(authResult = authResult, consentResponse = APPROVED_USER_VERIFICATION)
            return if (ctx.cibaEnabled && ctx.consentResponse == APPROVED_CONSENT_PROMPT) authedCtx.verify(exp).fold(
                { Result.success(Completed(challenge, authedCtx, it)) },
                { Result.failure(it) }
            ) else Result.success(UserConsent(challenge, authedCtx))
        }
```
cause if this from a ciba request we can call verify. and pass in exp.